addons/event_crm/models/event_lead_generate_request.py

```diff
@@ -0,0 +1,51 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import api, fields, models
+
+
+class EventLeadGenerateRequest(models.Model):
+    """ Technical model created when a user requests 'leads generation' on an event based on all
+    existing event.lead.rules (see event#action_generate_leads).
+
+    As an event can hold a lot of registrations, we use a batch approach with a separate model that
+    contains the batching logic method and the field to retain progress.
+
+    To benefit from a background processing, we use a CRON that calls itself with a CRON trigger
+    until the batch is completed, which unlinks this technical generation record. """
+
+    _name = "event.lead.generate.request"
+    _description = "Event Lead Generate Request"
```

Comment:
Don't forget _rec_name (event_id I suppose), and _order (to be deterministic when running jobs)